File: src/com/mycompany/myapp/entities/Contrat.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.mycompany.myapp.entities;

/**
 *
 * @author msi
 */
public class Contrat {
    
    private int id;
    private String montant;
    private String dateexpiration;
    private String imagecontrat;
    private String datesignature;

    public Contrat() {
    }

    public Contrat(int id, String montant, String dateexpiration, String imagecontrat, String datesignature) {
        this.id = id;
        this.montant = montant;
        this.dateexpiration = dateexpiration;
        this.imagecontrat = imagecontrat;
        this.datesignature= datesignature;
    }

    public Contrat(String montant, String dateexpiration, String imagecontrat, String datesignature) {
        this.montant = montant;
        this.dateexpiration = dateexpiration;
        this.imagecontrat = imagecontrat;
        this.datesignature = datesignature;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getMontant() {
        return montant;
    }

    public void setMontant(String montant) {
        this.montant = montant;
    }

    public String getDateexpiration() {
        return dateexpiration;
    }

    public void setDateexpiration(String dateexpiration) {
        this.dateexpiration = dateexpiration;
    }

    public String getImagecontrat() {
        return imagecontrat;
    }

    public void setImagecontrat(String imagecontrat) {
        this.imagecontrat = imagecontrat;
    }

    public String getDatesignature() {
        return datesignature;
    }

    public void setDatesignature(String datesignature) {
        this.datesignature = datesignature;
    }

}

File: src/Entity/Contrat.php
<?php

namespace App\Entity;

use App\Repository\ContratRepository;
use Doctrine\Common\Collections\ArrayCollection;
use Doctrine\Common\Collections\Collection;
use Doctrine\DBAL\Types\Types;
use App\Form\FileType;
use App\Entity\TypeContrat;
use Symfony\Component\Validator\Constraints as Assert;
use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Serializer\Annotation\Groups;


#[ORM\Entity(repositoryClass: ContratRepository::class)]
class Contrat
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    #[Groups("contrat")]
    private ?int $id = null;

    #[ORM\Column(type: Types::DATE_MUTABLE)]
    #[Assert\LessThanOrEqual("today UTC", message: "Date n est pa valable!")]
    #[Groups("contrat")]
    private ?\DateTimeInterface $datesignature = null;

    #[ORM\Column(type: Types::DATE_MUTABLE)]
    #[Assert\LessThanOrEqual("today UTC", message: "Date n est pa valable!")]
    #[Groups("contrat")]
    private ?\DateTimeInterface $dateexpiration = null;


    #[ORM\Column(length: 255)]
    #[Assert\NotBlank(message:'montant is required')]
    #[Groups("contrat")]
    private ?string $montant = null;

    #[ORM\Column(length: 255)]
    #[Groups("contrat")]
    private ?string $imagecontrat = null;
    #[Groups("contrat")]
    #[ORM\OneToMany(mappedBy: 'contrat', targetEntity: TypeContrat::class)]

    private Collection $nom;





    public function getId(): ?int
    {
        return $this->id;
    }

    public function getDatesignature(): ?\DateTimeInterface
    {
        return $this->datesignature;
    }

    public function setDatesignature(\DateTimeInterface $datesignature): self
    {
        $this->datesignature = $datesignature;

        return $this;
    }

    public function getDateexpiration(): ?\DateTimeInterface
    {
        return $this->dateexpiration;
    }

    public function setDateexpiration(\DateTimeInterface $dateexpiration): self
    {
        $this->dateexpiration = $dateexpiration;

        return $this;
    }


    public function getmontant(): ?string
    {
        return $this->montant;
    }

    public function setmontant(string $montant): self
    {
        $this->montant = $montant ;

        return $this;
    }

    public function __toString(): string
    {
        return $this->getDatesignature()->format('Y-m-d');
    }

    public function getImagecontrat(): ?string
    {
        return $this->imagecontrat;
    }

    public function setImagecontrat(string $imagecontrat): self
    {
        $this->imagecontrat = $imagecontrat ;

        return $this;
    }

    /**
     * @return Collection<int, TypeContrat>
     */
    public function getNom(): Collection
    {
        return $this->nom;
    }

    public function addNom(TypeContrat $nom): self
    {
        if (!$this->nom->contains($nom)) {
            $this->nom->add($nom);
            $nom->setContrat($this);
        }

        return $this;
    }

    public function removeNom(TypeContrat $nom): self
    {
        if ($this->nom->removeElement($nom)) {
            // set the owning side to null (unless already changed)
            if ($nom->getContrat() === $this) {
                $nom->setContrat(null);
            }
        }

        return $this;
    }


}
